src/d3a/models/market/market_structures.py

```diff
@@ -211,16 +203,16 @@ def __eq__(self, other):
                 self.requirements == other.requirements)
 
 
-def offer_or_bid_from_json_string(offer_or_bid, current_time=None):
+def offer_or_bid_from_json_string(offer_or_bid, current_time=None) -> Union[Offer, Bid]:
     offer_bid_dict = json.loads(offer_or_bid)
     object_type = offer_bid_dict.pop("type")
-    if "price" not in offer_bid_dict:
-        offer_bid_dict["price"] = offer_bid_dict["energy_rate"] * offer_bid_dict["energy"]
+
+    offer_bid_dict.pop("energy_rate", None)
+
     if object_type == "Offer":
-        offer_bid_dict.pop('energy_rate', None)
-        offer_bid_dict['time'] = current_time
+        offer_bid_dict["time"] = current_time
```

Comment:
Nit: I have the feeling that this logic doesn't belong here. But I see that you just took it from the existing version, so no need to fix it here.